app/jobs/setup_metadata_job.rb

```diff
@@ -14,6 +14,7 @@ def perform(parent_object, current_batch_process, current_batch_connection = par
       parent_object.processing_event("SetupMetadataJob failed to find all images.", "failed")
       return
     end
+    index_private(parent_object)
```

Comment:
Should we do the bother with the fetch and child jobs if the parent is private, @mikeapp?
If it's made visible or redirected, it will re-run setup_metadata_job and go through the normal process.